### ISSUE TITLE
Big jars are un-navigable

### ISSUE THREAD

**Issue by bedge** [recursean/JAR-Viewer-and-Decompiler-VSCode-Extension#4]
Big jars are hard to navigate as there's 
- no line numbers
- no search
Note where the scroll bar is on the right. There's 3000 lines in that window.

![image](https://github.com/user-attachments/assets/652cbebb-f1cc-40e7-9b47-2ae3b2d4ac82)

Aside from that, super handy. thanks!

**Comment by bedge** [comment]
Yep, wrong plugin...

**Comment by bedge** [comment]
I take it back, apparently all of the java decompiler extensions do this.

**Comment by recursean** [comment]
Hi Bedge,

Appreciate you trying out the extension. I am always trying to make the extension as useful as possible so thanks for opening this issue.

As you noted, the screenshot you provided in the original post is from another extension. This extension (where you opened this issue) displays the JAR contents in the file explorer window, which in your screenshot would be on the left side of VSCode. The extension you screenshotted puts the JAR contents in the main file editor window.

So, in regards to your first point of "no line numbers", I don't think it makes sense to add line numbers to the file explorer window. I personally have never seen line numbers in a file explorer before. I agree that for the other extension where the contents are displayed in the file editor window, it would maybe make sense to add line numbers there as line numbers are expected and normal in a file editor. But again, this extension does not display the contents in the file editor. Let me know if you feel differently from this.

In regards to "no search", I do agree that having some type of search or filter capability would be beneficial, especially for larger JAR files, like the one you are working with. A question I have for you is what exactly would you want to search on? Are you just looking to search for Java packages within the JAR file or are you looking to search by Java class? Or maybe something else?

Let me know your thoughts. Thanks again for trying out the extension.

**Comment by bedge** [comment]
Re: line numbers, I meant in the right side decompiled jar list. 
Reason being, to find the jar of interest I copy/pasted the jar list into another editor window and searched for it (given there is no search in the jar list window), found it on line 2027 and wanted to jump to that line in the jar list so I could open it.
So, granted, it's really only because there's no search that I wanted the line numbers.

For searching, anything in the class name, with either glob or regex support would be ideal.
Basically the maven groupId/artifactId.

Appreciate the response and looking forward to updates.

thanks!

**Comment by recursean** [comment]
I just released version [1.2.0](https://marketplace.visualstudio.com/items?itemName=seanmcshane.jar-viewer-and-decompiler) of this extension. This update adds a new search window where you can search for Java packages using either normal search strings or regular expressions. This essentially allows you to search and filter by the Maven group id. 

<img width="341" alt="image" src="https://github.com/user-attachments/assets/3f892112-6ada-4fb9-b339-4aee54aece5c">

I would love to hear your feedback! If you identify anything that could be improved and/or needs fixed, please let me know and I can address them in a future update.

**Comment by recursean** [comment]
Just checking in. If I don't hear back from you by next weekend, I will close this issue. Thanks!
